src/l500/l500-private.h

```diff
@@ -266,10 +266,10 @@ namespace librealsense
             using namespace std;
 
             auto table = reinterpret_cast<const T*>(raw_data.data());
-
-            if (raw_data.size() < sizeof(T))
+            auto expected_size = table->get_expected_size();
+            if (raw_data.size() < expected_size)
```

Comment:
What if we get more?

Reply:
If we get more we will use only the max_size data. (no out of bound exception like we have now)
I like it better when it's check equal, but this was the implementation so I did not touched it.